### ISSUE TITLE
increment file name does not work with zero prefixed file name

### ISSUE THREAD

**Issue by xh3b4sd** [microsoft/vscode#113809]
Version: 1.52.1
Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
Date: 2020-12-16T16:30:02.420Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 20.2.0

Steps to Reproduce:

1. Create a file called `0001004` in the explorer.
2. Copy that file. 
3. Paste that file. 
4. The new file name is `0001004.1`, where it should be `0001005`.

<img width="220" alt="Screenshot 2021-01-05 at 11 08 23" src="https://user-images.githubusercontent.com/552769/103634000-c832a780-4f46-11eb-9967-320c07e0219f.png">

Does this issue occur when all extensions are disabled?: Yes

**Comment by xh3b4sd** [comment]
I think the code is in https://github.com/microsoft/vscode/blob/ebdff39cca7f5ee5e4ddd16e92d7fce2cce5cd93/src/vs/workbench/contrib/files/browser/fileActions.ts#L363-L454. Last time I checked there was also no test case for that.

**Comment by isidorn** [comment]
Thanks. I acknowledge the issue. And we would welcome a PR that neatly fixes this and adds a test, thus adding help wanted label.

**Comment by xh3b4sd** [comment]
Not a Typescript magician here, otherwise I would have provided a pull request. 😬

**Comment by nrayburn-tech** [comment]
I can submit a PR.  @isidorn looking at the file/folder checks it looks like these two sets should be added.  If they look right, I will send a PR for them.

- File without extension
  - `file => file1`
  - `file1 => file2`
- Version number only, without extension
  - `1` => `2`

**Comment by isidorn** [comment]
@nrayburn-tech makes sense. Make sure to cover with tests and no need to go too fancy! Thanks

**Comment by nrayburn-tech** [comment]
@isidorn One other incrementing that I am not sure about.  `0999.txt` becomes `1000.txt`, without the leading 0.  Is this right, or should it maintain the number of leading 0s?

**Comment by xh3b4sd** [comment]
If you have zero leading paddings it usually means the amount of characters
should be maintained. So making 01000 out of 0999 would be wrong IMO. It
should just become 1000. Me two cents.

**Comment by isidorn** [comment]
Agree with @xh3b4sd

**Comment by isidorn** [comment]
@xh3b4sd it would be great if you can try this out with vscode insdiers from tomorrow and give us feedback.
Thanks to @nrayburn-tech for providing a PR

**Comment by xh3b4sd** [comment]
Will there be a release or how do I get my hands on the right version to test locally?

**Comment by isidorn** [comment]
https://code.visualstudio.com/insiders/ releases every day. Just get the release tomorrow and you should be good.

**Comment by xh3b4sd** [comment]
Tested the latest insider build. I could not break it. Works for me. Thanks a lot for the contribution. ❤️

**Comment by isidorn** [comment]
Thanks for letting us know. Adding verified label.
